tv_apps/smart-home/js/modal_dialog.js

```diff
@@ -0,0 +1,116 @@
+/* global SimpleKeyNavigation */
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * ModalDialog creates a dialog with a group of buttons as select options. It
+   * also includes open/close animation and bubbling animation on buttons.
+   * The template is shown below:
+   *   <smart-dialog class="modal-dialog">
+   *     <div class="outer-container">
+   *       <div class="container">
+   *         <div class="modal-dialog-message">
+   *         </div>
+   *         <div class="modal-dialog-button-group">
+   *           <smart-button></smart-button>
+   *           ...
+   *           <smart-button></smart-button>
+   *         </div>
+   *       </div>
+   *     </div>
+   *   </smart-dialog>
+   */
+  function ModalDialog() {
+    this.smartBubble = document.createElement('smart-bubbles');
+    document.body.appendChild(this.smartBubble);
+
+    this.element = document.createElement('smart-dialog');
+    this.element.classList.add('modal-dialog');
+    this.element.setAttribute('tabIndex', '-1');
+
+    // in order to make vertical align, we need additional container
+    this.outerContainer = document.createElement('div');
+    this.outerContainer.classList.add('outer-container');
+
+    this.container = document.createElement('div');
+    this.container.classList.add('container');
+
+    this.messageElement = document.createElement('div');
+    this.messageElement.classList.add('modal-dialog-message');
+
+    this.buttonGroup = document.createElement('div');
+    this.buttonGroup.classList.add('modal-dialog-button-group');
+    this.buttonGroup.setAttribute('smart-bubbles', 'true');
+
+    this.container.appendChild(this.messageElement);
+    this.container.appendChild(this.buttonGroup);
+
+    this.outerContainer.appendChild(this.container);
+    this.element.appendChild(this.outerContainer);
+
+    document.body.appendChild(this.element);
+
+    this.element.addEventListener('opened', function() {
+      this.smartBubble.play(this.buttonElements);
+    }.bind(this));
+
+    this.element.addEventListener('closed', function() {
+      this.element.classList.remove('visible');
+      if (this._clickedIndex &&
+          this.buttonSettings[this._clickedIndex].onClick) {
+        this.buttonSettings[this._clickedIndex].onClick();
+      } else if (this.onCancel) {
+        this.onCancel();
+      }
+    }.bind(this));
+
+    this.smartBubble.addEventListener('all-items-bubbled', function() {
```

Comment:
Please add more comments here to explain that we should start simple key navigator while all buttons show up.